tests/test_sources.py

```diff
@@ -307,6 +314,107 @@ def test_general_invalid_input():
     with pytest.raises(ValueError, match="Bounds do not form a plane."):
         Plane(bounds)
 
+def trace(
+    start_trace_nztm: np.ndarray[float], length: float, strike: float
+) -> np.ndarray:
+    # Do this in NZTM to prevent any issues with the coordinate system conversions
+    strike_vec = np.array([np.cos(np.radians(strike)), np.sin(np.radians(strike))])
+    end_trace_nztm = start_trace_nztm + strike_vec * length
+
+    trace_nztm = np.stack((start_trace_nztm, end_trace_nztm), axis=0)
+    return coordinates.nztm_to_wgs_depth(trace_nztm)
+
+
+@st.composite
+def valid_trace_definition(draw: st.DrawFn):
+    trace_point_1_nztm = draw(
+        st.builds(
+            nztm_coordinate,
+            y=st.floats(4715500, 6221500),
+            x=st.floats(1073000, 2154000),
+        )
+    )
+    trace_point_2_nztm = draw(
+        st.builds(
+            nztm_coordinate,
+            y=st.floats(4715500, 6221500),
+            x=st.floats(1073000, 2154000),
+        )
+    )
+    assume(not np.allclose(trace_point_1_nztm, trace_point_2_nztm))
+    trace_points_nztm = np.stack((trace_point_1_nztm, trace_point_2_nztm), axis=0)
+
+    # Compute strike
+    north_direction = np.array([1, 0, 0])
+    up_direction = np.array([0, 0, 1])
+    strike_vec = np.concatenate((trace_point_2_nztm - trace_point_1_nztm, [0]))
+    strike_nztm = geo.oriented_bearing_wrt_normal(
+        north_direction, strike_vec, up_direction
+    )
+
+    dtop = draw(st.floats(0, 100))
+    depth = draw(st.floats(1, 100))
+    dip = draw(st.floats(1, 90))
+
+    if np.isclose(dip, 90):
+        dip_dir_nztm = 0
+    else:
+        dip_dir_nztm = (strike_nztm + draw(st.floats(1, 179))) % 360
+
+    return (
+        trace_points_nztm,
+        dtop,
+        depth,
+        dip,
+        dip_dir_nztm,
+        strike_nztm,
+    )
+
+
+@given(valid_trace_definition())
+def test_plane_from_trace(data: tuple):
+    (
+        trace_points_nztm,
+        dtop,
+        depth,
+        dip,
+        dip_dir_nztm,
+        strike_nztm,
+    ) = data
+
+    plane = Plane.from_nztm_trace(trace_points_nztm, dtop, dtop + depth, dip, dip_dir_nztm)
+    assert np.isclose(plane.top_m, dtop * 1000, atol=1e-3)
+    assert np.isclose(plane.bottom_m, (dtop + depth) * 1000, atol=1e-3)
+    assert np.isclose(plane.dip, dip, atol=1e-6)
+    assert np.isclose(plane.dip_dir_nztm, dip_dir_nztm, atol=1e-3)
+    assert np.isclose(plane.strike_nztm, strike_nztm, atol=1e-6)
+    assert np.isclose(
+        plane.width, plane.projected_width / np.cos(np.radians(plane.dip)), atol=1e-6
+    )
+    assert np.allclose(
+        shapely.get_coordinates(plane.geometry, include_z=True)[:-1], plane.bounds
+    )
```

Comment:
Similarly here, you should (and I don't currently) use `np.testing.assert_array_almost_equal` for better test failure outputs.

```suggestion
    assert shapely.get_coordinates(plane.geometry, include_z=True)[:-1] == pytest.approx(plane.bounds)
```